File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Resources/readtimephasedresourcedata.rb
module Asposetasksjava
  module ReadTimephasedResourceData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

		# Get the Resource by its ID
		resource = project.getResources().getByUid(1)

		prj = Rjb::import('com.aspose.tasks.Prj')

		# Print Timephased data of ResourceWork
		puts "Timephased data of Resource Work"
		result = resource.getTimephasedData(project.get(prj.START_DATE), project.get(prj.FINISH_DATE)).toList()
		
		i = 0
		while i < result.size()	
			td = result.get(i)
		    puts "Start: " + td.getStart().toString()
		    puts "Work: " + td.getValue()
		    i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResources/ReadTimephasedResourceData.php
<?php
namespace Aspose\Tasks\WorkingWithResources;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Prj as Prj;

class ReadTimephasedResourceData{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

		# Get the Resource by its ID
		$resource = $project->getResources()->getByUid(1);

		$prj = new Prj();

		# Print Timephased data of ResourceWork
		print "Timephased data of Resource Work".PHP_EOL;
		$result = $resource->getTimephasedData($project->get($prj->START_DATE), $project->get($prj->FINISH_DATE))->toList();

		$i = 0;
		while ($i < sizeof($result)) {
            $td = $result -> get($i);
		    print "Start: " . (string)$td -> getStart().PHP_EOL;// . toString()
		    print "Work: " . (string)$td -> getValue().PHP_EOL;
		    $i += 1;
		}

    }


}
?>